frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -702,18 +722,22 @@ func (c *cvdCommand) Run() error {
 		// Need to call this once the work is done to release resources.
 		defer cancelFn()
 	}
+	combinedOut := &bytes.Buffer{}
+	stdOut := newFilterBuffer(combinedOut)
 	cmd := buildCvdCommand(ctx, c.execContext, c.androidHostOut, c.home, c.cvdBin, c.args...)
-	out, err := cmd.CombinedOutput()
+	cmd.Stdout = stdOut
+	cmd.Stderr = combinedOut
```

Comment:
IIUC this sets up stdout to write to two writers: `stdOut` and `combinedOut`, while stderr writes to `combinedOut` only.

I think this introduces a race:

`Writer`s are not routine-safe. `Command` guarantees that writes will be synchronized if `Stdout` and `Stderr` are the same `Writer` when compared with `==`. But by using a `filterBuffer` the precondition no longer holds.

Reply:
good catch!

used two different buffers instead.